File: lab2no2.rb
def illustration(banki, ostatok, count_steps)
  if count_steps == 0
    puts "Jars"
    puts "F" * banki
  elsif count_steps.odd?
    puts "#{count_steps} step"
    puts "E" * banki
  else
    puts "#{count_steps} step"
    puts "F" * (banki - ostatok)
    puts "E" * ostatok
  end
end

puts "Input jar quantity: "
banki = gets.chomp.to_i
puts "Input exchange: "
obmen = gets.chomp.to_i

if banki <= 0 || obmen <= 0
  puts "Error"
else
  count_empty = 0
  count_steps = 0
  ostatok = 0
  new_banki = banki

  while new_banki >= obmen
    if count_steps == 0
      illustration(new_banki, ostatok, count_steps)
    end
    count_steps += 1
    illustration(new_banki, ostatok, count_steps)
    count_empty += new_banki - ostatok
    ostatok = new_banki % obmen
    new_banki = (new_banki / obmen) + ostatok
    count_steps += 1
    illustration(new_banki, ostatok, count_steps)
  end

  if new_banki < obmen
    count_empty += new_banki - ostatok
    count_steps += 1
    illustration(new_banki, ostatok, count_steps)
    puts "Jars drunk: #{count_empty}"
    puts "Step count: #{count_steps}"
  end
end

File: lab2no2.go
package main

import (
	"fmt"
)

func illustration(banki, ostatok, countSteps int) {
	if countSteps == 0 {
		fmt.Println("Jars")
		for i := 0; i < banki; i++ {
			fmt.Print("F")
		}
		fmt.Println()
	} else if countSteps%2 != 0 {
		fmt.Printf("%d step\n", countSteps)
		for i := 0; i < banki; i++ {
			fmt.Print("E")
		}
		fmt.Println()
	} else {
		fmt.Printf("%d step\n", countSteps)
		for i := 0; i < (banki - ostatok); i++ {
			fmt.Print("F")
		}
		fmt.Println()
		for i := 0; i < ostatok; i++ {
			fmt.Print("E")
		}
		fmt.Println()
	}
}

func main() {
	var banki, obmen int

	fmt.Println("Input jar quantity: ")
	fmt.Scan(&banki)

	fmt.Println("Input exchange: ")
	fmt.Scan(&obmen)

	if banki <= 0 || obmen <= 0 {
		fmt.Println("Error")
	} else {
		countEmpty := 0
		countSteps := 0
		ostatok := 0
		newBanki := banki

		for newBanki >= obmen {
			if countSteps == 0 {
				illustration(newBanki, ostatok, countSteps)
			}
			countSteps++
			illustration(newBanki, ostatok, countSteps)
			countEmpty += newBanki - ostatok
			ostatok = newBanki % obmen
			newBanki = (newBanki / obmen) + ostatok
			countSteps++
			illustration(newBanki, ostatok, countSteps)
		}

		if newBanki < obmen {
			countEmpty += newBanki - ostatok
			countSteps++
			illustration(newBanki, ostatok, countSteps)
			fmt.Printf("Jars drunk: %d\n", countEmpty)
			fmt.Printf("Step count: %d\n", countSteps)
		}
	}
}
